### ISSUE TITLE
Additional documentation additions

### ISSUE THREAD

**Issue by multimokia** [EmitKnowledge/Signals#47]
Hi there!

I've been working with your framework and wanted to propose some additional clarification to the documentation. My first thoughts fall under the documentation for [Distributed Process](https://github.com/EmitKnowledge/Signals/wiki/1.3-Distributed-process), which could use a pitfalls section, especially noting that attempting to access `Context` from `Work` would throw a `NullReferenceException`

I think it'd also benefit from a data flow visual to help better imply the order of operations and some documentation noting which parts are run on what level of the app, for (i.e. that `Map` is called from the Process level and not background level)

I feel like additional documentation for things like `AutoApiProcess` could be documented too, same for fetching query parameters from `GET` requests and the shapes of `Context` at each level, as it appears to be different. Some sort of quick reference would go a long way.

I also feel like a significant improvement would be the ability to support custom HTTP codes in responses similarly to ASP.NET via the `StatusCode` param. Right now for both standard responses and return types, it seems like I'd need to inherit the base types and add the new property to support custom HTTP codes.

Thanks for reading, hoping this can help out future developers using this

**Comment by mnikolovski** [comment]
Hi there!

Thanks so much for your feedback! We’re thrilled to hear that someone outside our company is using Signals. 
You've raised some great points, and we’ll work on improving the documentation, including adding a pitfalls section and looking into better data flow visuals and custom HTTP status code support.

We appreciate you taking the time to share your thoughts - it will help make things better for everyone!

If I may ask, how satisfied are you with Signals? What kind of apps have you been building with it? Have you pushed it to production? What is the overall experience.
- I appreciate your time to provide us with feedback.

Best regards,
Marjan
CEO, EMIT KNOWLEDGE

**Comment by multimokia** [comment]
Hi Marjan,

Glad to hear we'll be seeing some improved documentation in the future, I'll probably keep posting or replying here as other issues arise or I dig more into the source for what we're working on.

At this time I'm working with a small team to develop a SaaS application -- nothing too big at this time and we're still working towards a release.

Overall it's been an interesting experience, a fairly different paradigm than ASP.NET for better and for worse, namely in the way that I feel like the architecture feels like it has gotten a little messy between the different "levels" of everything, namely in API, then `BusinessProcess`, which leads down further

if anything it feels like sometime the auth/validate system becomes a bit more of a hindrance especially when we're trying to treat the `BusinessProcess` layer as more of a services section, which would also help us make testing a little more agnostic due to more unit based code instead of more monolithic style processes.

Nothing we can't handle, but not always ideal.

Something else we were thinking is the ability to run the background process on a non-windows system or using another queuing system like RabbitMQ or the like. Instead of passing off to another process entirely, would it make sense to perhaps leverage a threadpool and offload the work to delegated processes executed there? That may reduce the complexity of the messaging system to a dedicated background service and also make OS agnosticism a little easier.

Something else we'd love to see coming from ASP is the ability to use async methods instead of synchronous ones. I understand that the background process is the answer to this, but we feel like keeping everything together would be nicer and easier to deploy

**Comment by multimokia** [comment]
I had another thought on something that could enhance the `result` type that is returned from processes. Since results contain `IsFaulted`, having some kind of `Exception` field and `Message` field could work, quite literally storing the exception that it encountered, but ignoring the datamember in API calls so sensitive information is still blocked out.

**Comment by mnikolovski** [comment]
Thank you for your feebdack. We will put this items into the backlog.
At the moment we are working on migrating Signals to .NET8(migrating .NETstandard to .NET8), upgrading the nuget packages to latest and all of the changes imposed by this change. It will be a major release.

**Comment by multimokia** [comment]
Awesome! I'd love to help if i'm able as my team is also looking to migrate up too.

**Comment by mnikolovski** [comment]
I would appreciate some assistance with testing. We’re currently working on the [upgrade-to-net8](https://github.com/EmitKnowledge/Signals/tree/upgrade-to-net8) branch, where we've completed the migration to .NET8. All NuGet packages have been updated to the latest versions, and deprecated packages have been removed.

Tests are running green, which is good so far.
![image](https://github.com/user-attachments/assets/589b99c2-0f2d-456c-80b2-113a209bf55f)

After we are done with testing, we will release the nugets to v3.0 on all packages. Ready to rock and roll :)
